### PR TITLE
[Fix] Enable ROD flag in API env example

### LINKED ISSUE #9547: ♻️ Set `FEATURE_RECORD_OF_DECISION` to true in API example
## ♻️ Debt/Refactor

Set RoD to true in `api/.env.example`.

image

## 🕵️ Details

The flag controls backend validation, it was disabled at the start as without frontend completion it would be impossible to complete tests in CI. Now it should be set to true to expose any issues with validation behind the flag. 

## ✅ Acceptance Criteria

- [ ] Flag on
- [ ] Issues that arise or failing tests either solved or spun off if complex

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] ticket number
```

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #9547 

## 👋 Introduction

Enabled the ROD feature flag in the API env example file and fixes the tests that were failing with it enabled.

## 🕵️ Details

The PHPUnit tests were extremely tricky to get to pass. I did not touch any actual assertions so it should be good but anyone reviewing should pay special attention there because I'm not super confident on them 😓  

## 🧪 Testing

1. Confirm `./api/.env.example` as the ROD flag turned on
2. Confirm tests still pass

### CODE REVIEW

**Comment by vd1992** [comment]
Ah yes, the skipped tests would be tricky to solve since they were broken by changes. 
`testAssessmentStepValidation` was written back when the screening questions step wasn't required to have skills attached

I'll look through and offer suggestions for test changes where I see them

**Review by vd1992 — CHANGES_REQUESTED** [review]
It is also a bit complicated by seeding not quite being correct, for example, seeding assessment plans incomplete. But that is outside scope and we already have issues to improve database seeding

Just some comments to improve

File: api/tests/Feature/PoolTest.php [hunk 1]
```diff
@@ -740,13 +738,6 @@ public function testAssessmentStepValidation(): void
...
         ]);
 
-        $step = AssessmentStep::factory()->create(
-            [
-                'pool_id' => $completePool->id,
-                'type' => AssessmentStepType::ADDITIONAL_ASSESSMENT->name,
-            ]
-        );
-
         // assert cannot publish due to the additional assessment step not having any pool skills
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/tests/Feature/PoolTest.php:741]: Okay, hopefully I got it right 😛 [update test comments to reflect new ROD factories](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9603/commits/8c46817d18a31ef6c215cc2261ad917e41339866)

File: api/tests/Feature/PoolTest.php [hunk 2]
```diff
@@ -763,8 +754,10 @@ public function testAssessmentStepValidation(): void
...
             ->assertGraphQLErrorMessage('AssessmentStepMissingSkills');
 
-        $step->type = AssessmentStepType::SCREENING_QUESTIONS_AT_APPLICATION->name;
-        $step->save();
+        $completePool->load(['assessmentSteps', 'poolSkills']);
+        foreach ($completePool->assessmentSteps as $assessmentStep) {
+            $assessmentStep->poolSkills()->sync($completePool->poolSkills->pluck('id')->toArray());
+        }
 
         // assert can publish if it is a screening questions step that is missing skills
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/tests/Feature/PoolTest.php]: Thanks for the suggestion, updated in [update test comments to reflect new ROD factories](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9603/commits/8c46817d18a31ef6c215cc2261ad917e41339866)

File: api/tests/Feature/PoolTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (vd1992) [reply api/tests/Feature/PoolTest.php:741]: Almost there, should delete line 741 as it is referencing lines 743-749 which were deleted

image

[the remark this answers is not in the document]
Comment (esizer) [reply api/tests/Feature/PoolTest.php]: oops! 😅  Removed in [remove old comment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9603/commits/d21db2e00b6851be476b0af5ed2b5706be5c5326)

**Review by vd1992 — APPROVED** [review]
Good stuff, nice!

**Pull request merged**